modules/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureClientProvider.java

```diff
@@ -317,6 +317,11 @@ private enum RetryMetricsTracker implements HttpPipelinePolicy {
 
         @Override
         public Mono<HttpResponse> process(HttpPipelineCallContext context, HttpPipelineNextPolicy next) {
+            if (requestIsPartOfABatch(context)) {
+                // Batch deletes fire once for each of the constituent requests, and they have a null response. Ignore those, we'll track
+                // metrics at the bulk level.
+                return next.process();
+            }
```

Comment:
It feels a bit odd that each individual deletion would go through the pipeline. But I assume that's just how azure sdk works.

Reply:
It actually might come in handy if we were batching billable actions. Delete is free so it doesn't matter, but the other action you can batch ("set blob tier") is not, and batches are billed per consituent request. If we were batching these we'd want to count those individual requests.